commit 8a18efab6eb6f0bfe220ba3644d438fd431bd529 (refs/changes/58/758/2)
Author: Trent Mick <trentm@gmail.com>
Date:   2016-10-20T16:58:43-07:00 (3 years ago)
    
    TOOLS-1584 imgapi client listImages() 'filters.channel' arg should win over 'this.channel'
    Reviewed by: Julien Gilli <julien.gilli@joyent.com>
    Approved by: Julien Gilli <julien.gilli@joyent.com>

diff --git a/lib/imgapi.js b/lib/imgapi.js
index e603170..4dbfdd7 100644
--- a/lib/imgapi.js
+++ b/lib/imgapi.js
@@ -563,10 +563,10 @@ IMGAPI.prototype.listImages = function listImages(filters, options, callback) {
 
     function listImagesWithLimit(headers, cb) {
         // limit and marker come straight from filters
-        var path = self._path('/images', filters, {
+        var path = self._path('/images', {
             channel: self.channel,
             inclAdminFields: inclAdminFields
-        });
+        }, filters);
         var reqOpts = {
             path: path,
             headers: headers
@@ -609,10 +609,10 @@ IMGAPI.prototype.listImages = function listImages(filters, options, callback) {
                 filters.limit = limit;
             }
 
-            var path = self._path('/images', filters, {
+            var path = self._path('/images', {
                 channel: self.channel,
                 inclAdminFields: inclAdminFields
-            });
+            }, filters);
             var reqOpts = {
                 path: path,
                 headers: headers
